rust/lance-table/src/io/commit.rs

```diff
@@ -60,6 +60,94 @@ use crate::format::{Index, Manifest};
 const VERSIONS_DIR: &str = "_versions";
 const MANIFEST_EXTENSION: &str = "manifest";
 
+/// How manifest files should be named.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum ManifestNamingScheme {
+    /// `_versions/{version}.manifest`
+    V1,
+    /// `_manifests/{u64::MAX - version}.manifest`
+    ///
+    /// Zero-padded and reversed for O(1) lookup of latest version on object stores.
+    V2,
+}
+
+impl ManifestNamingScheme {
+    pub fn manifest_path(&self, base: &Path, version: u64) -> Path {
+        let directory = base.child(VERSIONS_DIR);
+        match self {
+            Self::V1 => directory.child(format!("{version}.{MANIFEST_EXTENSION}")),
+            Self::V2 => {
+                let inverted_version = u64::MAX - version;
+                directory.child(format!("{inverted_version:020}.{MANIFEST_EXTENSION}"))
```

Comment:
I hope to live to the day where the fact that this is zero padded becomes significant.

Reply:
I hadn’t thought of that but lol.